app/Helpers/database/user-relationship.php

```diff
@@ -71,69 +71,70 @@ function changeFriendStatus(User $senderUser, User $targetUser, int $newStatus):
     }
 }
 
-function isUserBlocking(string $user, ?string $possibly_blocked_user): bool
+function GetFriendList(User $user): array
 {
-    if (!isset($possibly_blocked_user)) {
-        return false;
-    }
-
-    return GetFriendship($user, $possibly_blocked_user) == UserRelationship::Blocked;
-}
-
-function GetFriendship(string $user, string $friend): int
-{
-    sanitize_sql_inputs($user, $friend);
-
-    $query = "SELECT Friendship FROM Friends WHERE User='$user' AND Friend='$friend'";
-    $dbresult = s_mysql_query($query);
-    if ($dbresult) {
-        $data = mysqli_fetch_assoc($dbresult);
-        if ($data) {
-            return (int) $data['Friendship'];
-        }
-    }
-
-    return UserRelationship::NotFollowing;
+    $friendList = $user->followedUsers()
+        ->where('Permissions', '>=', Permissions::Unregistered)
+        ->whereNull('Deleted')
+        ->orderBy('LastActivityID', 'DESC')
+        ->get()
+        ->map(function ($friend) {
+            return [
+                'Friend' => $friend->User,
+                'RAPoints' => $friend->points,
+                'LastSeen' => empty($friend->RichPresenceMsg) ? 'Unknown' : strip_tags($friend->RichPresenceMsg),
```

Comment:
The old logic appears to have been converting empty and "Unknown" to `_`. I'm not sure why - this seems better and is consistent with `GetExtendedFriendsList`.